toolkit/tools/pkg/imagecustomizerlib/resolvconf.go

```diff
@@ -0,0 +1,143 @@
+	logger.Log.Infof("Overriding resolv.conf file")
+
+	imageResolveConfPath := filepath.Join(imageChroot.RootDir(), resolvConfPath)
+
+	existing := resolvConfInfo{}
+
+	stat, err := os.Lstat(imageResolveConfPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return resolvConfInfo{}, fmt.Errorf("failed to stat resolv.conf file:\n%w", err)
+		}
+		existing.existingType = resolvConfTypeNone
+	} else if stat.Mode()&os.ModeSymlink != 0 {
+		symlinkPath, err := os.Readlink(imageResolveConfPath)
+		if err != nil {
+			return resolvConfInfo{}, fmt.Errorf("failed to read resolv.conf symlink's path:\n%w", err)
+		}
+		existing.existingType = resolvConfTypeSymlink
+		existing.symlinkPath = symlinkPath
+	} else {
+		fileContents, err := file.Read(imageResolveConfPath)
+		if err != nil {
+			return resolvConfInfo{}, fmt.Errorf("failed to read resolv.conf file:\n%w", err)
+		}
+		existing.existingType = resolvConfTypeFile
+		existing.fileContents = fileContents
+		existing.filePerms = stat.Mode().Perm()
+	}
+
+	// Remove the existing resolv.conf file, if it exists.
+	// Note: It is assumed that the image will have a process that runs on boot that will override the resolv.conf
+	// file. For example, systemd-resolved. So, it isn't neccessary to make a back-up of the existing file.
+	err = os.RemoveAll(imageResolveConfPath)
+	if err != nil {
+		return resolvConfInfo{}, fmt.Errorf("failed to delete existing resolv.conf file:\n%w", err)
+	}
+
+	err = file.Copy(resolvConfPath, imageResolveConfPath)
+	if err != nil {
+		return resolvConfInfo{}, fmt.Errorf("failed to override resolv.conf file with host's resolv.conf:\n%w", err)
+	}
+
+	return existing, nil
+}
+
+func restoreResolvConf(existing resolvConfInfo, imageChroot *safechroot.Chroot) error {
+	logger.Log.Infof("Restoring resolv.conf")
+
+	imageResolveConfPath := filepath.Join(imageChroot.RootDir(), resolvConfPath)
+
+	// Delete the overridden resolv.conf file.
+	err := os.RemoveAll(imageResolveConfPath)
+	if err != nil {
+		return fmt.Errorf("failed to delete overridden resolv.conf file:\n%w", err)
+	}
+
+	switch existing.existingType {
+	case resolvConfTypeNone:
+		// Check if systemd-resolved is enabled.
+		resolvedEnabled, err := systemd.IsServiceEnabled("systemd-resolved.service", imageChroot)
+		if err != nil {
+			return err
+		}
+
+		if resolvedEnabled {
+			logger.Log.Infof("Adding systemd-resolved resolv.conf symlink")
+
+			// The systemd-resolved.service is enabled.
+			// So, create the symlink for the resolv.conf file.
+			// While technically this symlink will be automatically created on first boot, doing
+			// it now is helpful for verity images where the root filesystem is readonly.
+			err := os.Symlink(resolvSystemdStubPath, imageResolveConfPath)
+			if err != nil {
+				return fmt.Errorf("failed to create resolv.conf symlink:\n%w", err)
+			}
+		}
+
+	case resolvConfTypeFile:
+		logger.Log.Debugf("Restoring resolv.conf file")
```

Comment:
Right now, for the `resolvConfTypeNone` scenario, we add an additional info-level log that we're adding the symlink as part of restoring the resolv.conf file. For the other two options we're only doing a debug-level log. I think it would be useful to stay consistent. Personally, I'm leaning towards making the debug-level logs and info and removing the info log at the very beginning of the function, but my main focus is on consistency.

Reply:
Firstly, I think it is goodness to have an info log before anything happens. In this case, before the overridden `resolv.conf` file is removed.

Secondly, I think that restoring the previous file/symlink falls under banner of "Restoring resolv.conf". But creating a new symlink probably deserves a special callout. Hence the additional info log.